### ISSUE TITLE
Saver unable to restore from checkpoint file although it had successfully restored previously.

### ISSUE THREAD

**Issue by kwotsin** [tensorflow/tensorflow#7244]
I am currently seeing a very strange behavior from using tf.train.Saver. Basically I had previously successfully run my code and restored all the variables from the TensorFlow-slim inception-resnet-v2 model and even obtained my losses and predictions. However, after I added some scopes to exclude in `variables_to_restore = slim.get_variables_to_restore(exclude = [...])` , which I successfully run,  when I returned back to using no excluded scope, my code actually fails. It gave me the following error:

`NotFoundError (see above for traceback): Tensor name "InceptionResnetV2/Repeat_1/block17_13/Branch_1/Conv2d_0c_7x1/BatchNorm/beta/Adam" not found in checkpoint files ./inception_resnet_v2_2016_08_30.ckpt`

Here is the code I run.
```
with slim.arg_scope(inception_resnet_v2.inception_resnet_v2_arg_scope()):
	logits, end_points = inception_resnet_v2.inception_resnet_v2(image_batch, is_training = False)
	predictions = end_points['Predictions']

#....then the typical building of operations till you get the train_op

variables_to_restore = slim.get_variables_to_restore()
saver = tf.train.Saver(variables_to_restore) #This is the line where the errors appear
def restore_fn(sess):
	return saver.restore(sess, checkpoint_file)
sv = tf.train.Supervisor(logdir = log_dir, summary_op = None, init_fn = restore_fn)
```

I have even replaced my checkpoint file again and this error persists. Why does this happen?

I strongly suspect it might have been because the original graph has been edited again as I did not encapsulate most of the definitions within a specified graph like `with tf.Graph.as_default() as g:`, but at the start of my code I have always done `tf.reset_default_graph()`.

**Comment by yaroslavvb** [comment]
If you think it's a bug in TensorFlow you can use [inspect_checkpoint](https://github.com/tensorflow/tensorflow/blob/ec7929b878926c39255254e9aea992f0bc65aa68/tensorflow/python/tools/BUILD#L54) tool to isolate it further. IE, check what names are saved in checkpoint, and then what names are getting restored (`print tf.get_default_graph().as_graph_def()` will let you obtain you names of variables in current graph)

**Comment by kwotsin** [comment]
@yaroslavvb After inspecting the checkpoint, I have found 908 variables to restore, but using `slim.get_variables_to_restore()` I got around 1.5k variables. Turns out it was because I defined the variables to restore after I created my train_op, which caused many more variables that weren't defined in the checkpoint file to be tried to restore. After I defined the variables to restore **immediately after** my model inference, the problem is now solved. Thank you for your help!
